### PR TITLE
[WOR-920] Azure workspace creation test

### PR DESCRIPTION
**Pull request by cahrens**
Ticket: https://broadworkbench.atlassian.net/browse/WOR-920

Successful test run: https://github.com/broadinstitute/rawls/actions/runs/4953204483

````
23:01:54.872 [io-compute-blocker-2] INFO org.broadinstitute.dsde.workbench.fixture.BillingFixtures$ - Billing project tmp-billing-project-4422255a96 created.
23:01:55.251 [io-compute-blocker-2] INFO org.broadinstitute.dsde.workbench.service.Rawls$ - Creating workspace: tmp-billing-project-4422255a96/536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace authDomain: Set()
[WARN] [05/11/2023 23:02:12.453] [default-akka.actor.default-dispatcher-5] [akka.actor.ActorSystemImpl(default)] Illegal header: Illegal 'location' header: Invalid input '^', expected 'EOI', '#', '?', '@', lower-reg-name-char, path-abempty, pct-encoded, userinfo-char, ':' or UPPER_ALPHA (line 1, column 56): [https://rawls.rawls-4953204483-1-dev.bee.envs-terra.bio^(?!/introspect/)(/)(.*)workspaces/tmp-billing-project-4422255a96/536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace](https://rawls.rawls-4953204483-1-dev.bee.envs-terra.bio%5E%28/?!/introspect/)(/)(.*)workspaces/tmp-billing-project-4422255a96/536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace)
                                                       ^
23:02:12.457 [default-akka.actor.default-dispatcher-10] DEBUG org.broadinstitute.dsde.workbench.service.Rawls$ - API request: POST https://rawls.rawls-4953204483-1-dev.bee.envs-terra.bio/api/workspaces with headers (Authorization) {"namespace":"tmp-billing-project-4422255a96","name":"536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace","attributes":{"disableAutomaticAppCreation":"true"},"authorizationDomain":[]}
API response: 201 Created {"attributes":{"disableAutomaticAppCreation":"true"},"authorizationDomain":[],"bucketName":"","createdBy":"hermione.owner@quality.firecloud.org","createdDate":"2023-05-11T23:01:55.643Z","googleProject":"","isLocked":false,"lastModified":"2023-05-11T23:01:55.643Z","name":"536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace","namespace":"tmp-billing-project-4422255a96","workspaceId":"a21c3b82-c119-4d74-88b9-e3a07bab1349","workspaceType":"mc","workspaceVersion":"v2"}
23:02:12.457 [io-compute-blocker-2] INFO org.broadinstitute.dsde.workbench.service.Rawls$ - Getting workspace 536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace in tmp-billing-project-4422255a96
23:02:13.900 [default-akka.actor.default-dispatcher-11] DEBUG org.broadinstitute.dsde.workbench.service.Rawls$ - API request: GET https://rawls.rawls-4953204483-1-dev.bee.envs-terra.bio/api/workspaces/tmp-billing-project-4422255a96/536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace with headers (Authorization) 
API response: 200 OK {"azureContext":{"managedResourceGroupId":"staticTestingMrg","subscriptionId":"f557c728-871d-408c-a28b-eb6b2141a087","tenantId":"fad90753-2022-4456-9b0a-c7e5b934e408"},"catalog":false,"workspaceSubmissionStats":{"runningSubmissionsCount":0},"accessLevel":"OWNER","owners":["hermione.owner@quality.firecloud.org"],"workspace":{"attributes":{"disableAutomaticAppCreation":"true"},"authorizationDomain":[],"bucketName":"","cloudPlatform":"Azure","createdBy":"hermione.owner@quality.firecloud.org","creat...
23:02:14.047 [io-compute-blocker-2] INFO org.broadinstitute.dsde.workbench.service.Rawls$ - Deleting workspace: tmp-billing-project-4422255a96/536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace
23:02:17.611 [default-akka.actor.default-dispatcher-19] DEBUG org.broadinstitute.dsde.workbench.service.Rawls$ - API request: DELETE https://rawls.rawls-4953204483-1-dev.bee.envs-terra.bio/api/workspaces/tmp-billing-project-4422255a96/536e5537-2577-4d95-b2da-1541c76bd670-azure-test-workspace with headers (Authorization) null
API response: 202 Accepted Your workspace has been deleted.

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspacesAzureApiSpec.scala [hunk 1]
```diff
@@ -35,24 +35,23 @@ class AzureWorkspacesSpec extends AnyFlatSpec with Matchers with CleanUp {
...
-//      Rawls.workspaces.delete(projectName, workspaceName)
-//      assertNoAccessToWorkspace(projectName, workspaceName)
-//    }
+    withTemporaryAzureBillingProject(azureManagedAppCoordinates) { projectName =>
+      val workspaceName = generateWorkspaceName()
+      Rawls.workspaces.create(
+        projectName,
+        workspaceName,
+        Set.empty,
+        Map(AttributeName.withDefaultNS("disableAutomaticAppCreation") -> AttributeBoolean(true))
```
Comment (cahrens) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspacesAzureApiSpec.scala:44]: @marctalbott will change to

`Map("disableAutomaticAppCreation" -> "true")`

Comment (cahrens) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspacesAzureApiSpec.scala]: And will need to change this as well: https://github.com/broadinstitute/rawls/pull/2339/files#diff-518dd904cf9703f8db8998f3dd2eab4c5ef7b5488d59532eb194a391370e63e7R586

File: .github/workflows/rawls-run-azure-e2e-tests.yaml [hunk 2]
```diff
@@ -8,6 +8,11 @@ on:
         required: true
         default: 'develop'
         type: string
+      delete-bee:
```
Comment (cahrens) [inline .github/workflows/rawls-run-azure-e2e-tests.yaml:11]: Useful to leave the bee around for debugging test issues.

image

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala [hunk 3]
```diff
@@ -582,7 +583,7 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
                                       workspaceAttributeMap: AttributeMap
   ): Future[Unit] =
     workspaceAttributeMap.get(AttributeName.withDefaultNS("disableAutomaticAppCreation")) match {
-      case Some(AttributeBoolean(true)) =>
+      case Some(AttributeString("true")) | Some(AttributeBoolean(true)) =>
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala:586]: Decided to be tolerant of either "true" or `true`.

**Comment by cahrens** [comment]
jenkins retest

**Review by aherbst-broad — APPROVED** [review]
Thanks for doing this.

**Review by marctalbott — APPROVED** [review]

**Pull request merged**